backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceMetrics.java

```diff
@@ -0,0 +1,355 @@
+
+    /**
+     * Maximum time between requests in milliseconds
+     */
+    public final static int maxTimeBetweenRetriesMs = 5 * 60 * 1000;
+
+    /**
+     * Default submission url
+     */
+    public final static String defaultBaseUrl = "https://events.backtrace.io/api";
+
+    private static final transient String LOG_TAG = BacktraceMetrics.class.getSimpleName();
+
+    /**
+     * Default unique event name that will be generated on app startup
+     */
+    public final String defaultUniqueEventName = "guid";
+
+    /**
+     * Name of the summed event that will be generated on app startup
+     */
+    private final String startupSummedEventName = "Application Launches";
+
+    /**
+     * Unique Events handler
+     */
+    public UniqueEventsHandler uniqueEventsHandler;
+
+    /**
+     * Summed Events handler
+     */
+    public SummedEventsHandler summedEventsHandler;
+
+    /**
+     * Custom attributes provided by the user to BacktraceBase
+     */
+    protected Map<String, Object> customReportAttributes;
+
+    /**
+     * The application context
+     */
+    protected Context context;
+
+    /**
+     * Backtrace metrics settings
+     */
+    protected BacktraceMetricsSettings settings = null;
+
+    /**
+     * Name of the unique event that will be generated on app startup
+     */
+    private String startupUniqueEventName = defaultUniqueEventName;
+
+    /**
+     * Maximum number of events in store. If number of events in store hit the limit
+     * BacktraceMetrics instance will send data to Backtrace.
+     */
+    private int maximumNumberOfEvents = 350;
+
+    /**
+     * User custom request method
+     */
+    private final RequestHandler requestHandler = null;
+
+    /**
+     * Backtrace API class for metrics sending
+     */
+    private final Api backtraceApi;
+
+    /**
+     * Create new Backtrace metrics instance
+     *
+     * @param context                Application context
+     * @param customReportAttributes Backtrace client custom report attributes (must be nonnull)
+     * @param backtraceApi           Backtrace API for metrics sending
+     */
+    public BacktraceMetrics(Context context, @NonNull Map<String, Object> customReportAttributes, Api backtraceApi) {
+        this.context = context;
+        this.customReportAttributes = customReportAttributes;
+        this.backtraceApi = backtraceApi;
+    }
+
+    /**
+     * Enable metrics
+     *
+     * @param settings for Backtrace metrics
+     */
+    public void enable(BacktraceMetricsSettings settings) {
+        this.settings = settings;
+        // For tracking crash-free sessions we need to add
+        // application.session and application.version to Backtrace attributes
+        String sessionId = UUID.randomUUID().toString();
+        this.customReportAttributes.put("application.session", sessionId);
+        try {
+            this.customReportAttributes.put("application.version", this.context.getPackageManager()
```

Comment:
I think we should grab this attribute from BacktraceAttributes implementation (we already have a function there)

Reply:
Ah sorry I forgot to change this to `BacktraceAttributes` implementation, will do thank you!